File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/FizzBuzz.java
package fizzbuzz;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

public class FizzBuzz {
  public static final int MAX_NUMBER = 100;
  private FizzBuzzList _list;
  private FizzBuzzType _type;

  public FizzBuzz(FizzBuzzType type) {
    List<FizzBuzzValue> list = new ArrayList<>();
    _list = new FizzBuzzList(list);
    _type = type;
  }

  public FizzBuzzList generateList() {
    IntStream.rangeClosed(1, MAX_NUMBER).forEach(
      number -> _list = _list.add(generate(number))
    );
    return _list;
  }

  public FizzBuzzValue generate(int number) {
    return _type.generate(number);
  }

}

File: dev/20190822/k2works/src/FizzBuzz.ts
import { FizzBuzzValue } from "./FizzBuzzValue";
import { IFizzBuzzType } from "./IFizzBuzzType";

export class FizzBuzz {
  public get list(): FizzBuzzValue[] {
    return this._list;
  }
  public static readonly MAX_NUMBER: number = 100;
  public static readonly range = (start: number, end: number) =>
    Array.from({ length: end - start + 1 }, (v, k) => k + start);

  // tslint:disable-next-line: variable-name
  private _list: FizzBuzzValue[];
  // tslint:disable-next-line: variable-name
  private _type: IFizzBuzzType;

  constructor(type: IFizzBuzzType) {
    this._type = type;
  }

  public generate(input: number): FizzBuzzValue {
    return this._type.generate(input);
  }

  public generateList(): void {
    this._list = FizzBuzz.range(1, FizzBuzz.MAX_NUMBER).map(i =>
      this.generate(i)
    );
  }
}
